src/main/java/cam72cam/immersiverailroading/entity/EntityRollingStock.java

```diff
@@ -69,6 +92,30 @@ public String getDefinitionID() {
 		return this.defID;
 	}
 	
+	public LockType getLockType () {
+		return LockType.valueOf(this.getDataManager().get(LOCK_TYPE));
+	}
+	
+	public void setLockType (LockType type) {
+		this.getDataManager().set(LOCK_TYPE, type.toString());
+	}
+	
+	public UUID getLockOwner() {
+		if (this.getDataManager().get(LOCK_OWNER) == "") {
+			return null;
+		}
+		return UUID.fromString(this.getDataManager().get(LOCK_OWNER));
+	}
+	
+	public void setLockOwner(String playerId) {
+		this.getDataManager().set(LOCK_OWNER, playerId);
+	}
+	
+	public void switchLockType (EntityPlayer player) {
+		setLockType(getLockType().next());
```

Comment:
Send the player a message when this happens so they know what lock level this is